src/torchmetrics/detection/mean_ap.py

```diff
@@ -630,29 +647,43 @@ def _get_coco_datasets(self, average: Literal["macro", "micro"]) -> tuple[object
 
         return coco_preds, coco_target
 
-    def _coco_stats_to_tensor_dict(self, stats: list[float], prefix: str) -> dict[str, Tensor]:
+    def _coco_stats_to_tensor_dict(self, stats: list[float], prefix: str, iou_type: list) -> dict[str, Tensor]:
         """Converts the output of COCOeval.stats to a dict of tensors."""
         mdt = self.max_detection_thresholds
+
+        if set(iou_type).intersection(("mask", "bbox")):
+            return {
+                f"{prefix}map": torch.tensor([stats[0]], dtype=torch.float32),
+                f"{prefix}map_50": torch.tensor([stats[1]], dtype=torch.float32),
+                f"{prefix}map_75": torch.tensor([stats[2]], dtype=torch.float32),
+                f"{prefix}map_small": torch.tensor([stats[3]], dtype=torch.float32),
+                f"{prefix}map_medium": torch.tensor([stats[4]], dtype=torch.float32),
+                f"{prefix}map_large": torch.tensor([stats[5]], dtype=torch.float32),
+                f"{prefix}mar_{mdt[0]}": torch.tensor([stats[6]], dtype=torch.float32),
+                f"{prefix}mar_{mdt[1]}": torch.tensor([stats[7]], dtype=torch.float32),
+                f"{prefix}mar_{mdt[2]}": torch.tensor([stats[8]], dtype=torch.float32),
+                f"{prefix}mar_small": torch.tensor([stats[9]], dtype=torch.float32),
+                f"{prefix}mar_medium": torch.tensor([stats[10]], dtype=torch.float32),
+                f"{prefix}mar_large": torch.tensor([stats[11]], dtype=torch.float32),
+            }
```

Comment:
it should be possible to write this a bit more efficient if the only difference between the two is that `_small` has been removed for keypoint detection